### DISCUSSION TITLE
How to enable longhorn storage for terraform-managed EKS cluster

### DISCUSSION THREAD

**Discussion by eduard93** [longhorn/longhorn#2654] [Q&A]
Hello.

I have a deployed EKS cluster, created and managed using Terraform. It looks something like [this](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster):

```
resource "aws_eks_cluster" "example" {
  name     = "example"
  role_arn = aws_iam_role.example.arn

  vpc_config {
    subnet_ids = [aws_subnet.example1.id, aws_subnet.example2.id]
  }
}
```

My question is - what's the best way to install longhorn on this Kubernetes cluster automatically?

**Comment by jenting** [comment]
Longhorn is an application on top of Kubernetes, the current way we provided to install Longhorn is by Rancher Catalog, helm, or kubectl.
If you want to install longhorn automatically, probably you need to consider using the GitOps way to automatically install Longhorn on the Kubernetes cluster.

**Reply by eduard93** [reply]
Thank you for your reply!

So essentially whatever calls terraform should also call kubectl/helm/Rancher?

**Reply by jenting** [reply]
Yes. However, I don't know what's the best way to use terraform to call kubectl/helm/Rancher.
